### ISSUE TITLE
[DataGridPro] When I made columns toggle on/off for Columns in DataGrid Toolbar, scroll is  re-render problem

### ISSUE THREAD

**Issue by atillaaliyev** [mui/mui-x#3836]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When I made columns toggle on/off for Columns in DataGrid Toolbar, scroll is re-rendering. Maybe, I will hide more than one columns, but every time I have to down with scroll.

https://user-images.githubusercontent.com/6121871/152485679-e70ca6bf-e7bb-4861-ac13-91c0e19280a1.mp4

****

### Expected behavior 🤔

Scroll shouldn't re-render when toggle on/off for columns every time.

### Steps to reproduce 🕹

Steps:

1.
2.
3.
4.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID 💳 (optional)

37219

**Comment by cherniavskii** [comment]
Hey @atillaaliyev 
I've created a codesandbox example, but I cannot reproduce the issue.
https://codesandbox.io/s/toolbargrid-material-demo-forked-8bcsx?file=/demo.tsx

Can you create a minimal example reproducing the issue? That would help a lot!

**Comment by atillaaliyev** [comment]
https://codesandbox.io/s/bv9kt

You scroll to the most bottom and hide/show for column 8. You will notice it.

**Comment by cherniavskii** [comment]
@atillaaliyev Thanks, i can reproduce it now!
I had to add more columns to the grid, so that columns list is longer.
Here's a codesandbox with a single grid reproducing the issue: https://codesandbox.io/s/toolbargrid-material-demo-forked-gn5e2?file=/demo.tsx

**Comment by alexfauquette** [comment]
This issue has been solved by #3975

In locale, I can reproduce the bug up to the commit of this PR

Here is the reproduction case using the DataGridPro generated by the PR:
https://codesandbox.io/s/create-react-app-with-typescript-forked-2hn1i9?file=/src/App.tsx

@atillaaliyev Don't hesitate to reopen the issue if you encounter the same problem after the next release
